### ISSUE TITLE
Cypress tests fail abut Gitlab CI Exit successfully

### ISSUE THREAD

**Issue by yuanyuanlimaggie** [cypress-io/cypress#4626]
### Current behavior:

Our tests failed but in Gitlab CI, it doesn't exit with correct error code. It still says `Job succeeded`

![image](https://user-images.githubusercontent.com/44100783/60438903-b8af0180-9bdf-11e9-870e-a9d8e742ab3e.png)

### Desired behavior:

The job should failed with `ERROR: Job failed: `

### Versions

Operating System: Gitlab CI
Cypress Version: 3.3.2
Browser Version: Electron headless 
Base Image: 12.1.0

**Comment by jennifer-shehane** [comment]
Are you using a custom reporter with Cypress? https://on.cypress.io/reporters If so, can you provide the details around your reporter setup?

Does this always happen on every run?

**Comment by yuanyuanlimaggie** [comment]
Hi, Jennifer.

Thanks for the reply. I don’t use any custom reporter. It works well before, it started to not exit correctly a while ago. If all tests passed, it will be fine. But when any tests failed, it sometimes just showing “Job Successfully” and the gitlab CI will be in a green and tests pass status.

I don’t know if it’s helpful, our tests failed now also because of Chromium crash problem. So if the Chromium renderer is not crashed, all tests are good. But once it crashed, most of time it will exit with “Job succeeded”.
shttps://docs.cypress.io/guides/references/error-messages.html#The-Chromium-Renderer-process-just-crashed

**Comment by yuanyuanlimaggie** [comment]
Hi,
To add more information about the exit problem. Some job even exit earlier when no error shows yet. See the screenshot for Gitlab CI console.

Best
Yuanyuan


From: Yuanyuan Li
Sent: Tuesday, July 2, 2019 9:23 AM
To: 'cypress-io/cypress' <reply@reply.github.com>; cypress-io/cypress <cypress@noreply.github.com>
Subject: RE: [cypress-io/cypress] Cypress tests fail abut Gitlab CI Exit successfully (#4626)

Hi, Jennifer.

Thanks for the reply. I don’t use any custom reporter. It works well before, it started to not exit correctly a while ago. If all tests passed, it will be fine. But when any tests failed, it sometimes just showing “Job Successfully” and the gitlab CI will be in a green and tests pass status.

I don’t know if it’s helpful, our tests failed now also because of Chromium crash problem. So if the Chromium renderer is not crashed, all tests are good. But once it crashed, most of time it will exit with “Job succeeded”.
shttps://docs.cypress.io/guides/references/error-messages.html#The-Chromium-Renderer-process-just-crashed

**Comment by yuanyuanlimaggie** [comment]
@jennifer-shehane Thanks for your help!

Here is the gitlab configuration:

<img width="1092" alt="Screen Shot 2019-07-08 at 9 37 00 AM" src="https://user-images.githubusercontent.com/44100783/60814666-07bae080-a164-11e9-9a8d-3d9fb71ea3d9.png">

The base image we use is a docker image:

```
FROM cypress/base:12.1.0

RUN yarn global add cypress@3.3.2
```

**Comment by jordandlaman** [comment]
Can confirm this is now happening for my team as well in gitlab.

Just bumped the docker image from the 8.x.x chrome 74 to the cypress/browser node 12.x.x chrome 75 version and the tests failed but the job passed.

**Comment by magicgrl111** [comment]
I work with @yuanyuanlimaggie and we've also experienced the same early exits after updating to cypress@3.4.0

Our current base image: 
```
FROM cypress/base:12.1.0

RUN yarn global add cypress@3.4.0
```

**Comment by briancaffey** [comment]
I'm having the same issue when testing GitLab CI jobs locally with GitLab runner: Cypress exits early after finishing all tests with no error code.

**Comment by devsh4** [comment]
I'm facing a similar issue when running with docker image ```cypress/included:3.4.0``` in AWS CodePipeline. The tests fail however the build succeeds.

**Comment by Jakegillingham5** [comment]
Also facing a similar issue in GitLab where it is closing the job prior to running any tests. Screen shot attached. Wondering if @yuanyuanlimaggie | @magicgrl111 worked out a solution to the premature job closes.

![screen_shot](https://user-images.githubusercontent.com/11712293/65002479-2ba05c80-d933-11e9-8a6a-0f68b839c2ca.jpg)

**Comment by NicolasHofmairT23** [comment]
@Jakegillingham5 @briancaffey  I am experiencing a similiar behaviour. "Running: ..." and after a while just "Job succeeded". Even when the first executed test failed. Did you find a solution for this problem?

**Comment by makris55** [comment]
I also had this problem and finally I think we ended up solving it.
We have our own gitalb-runner so we can modify the config.toml and that's where we could solve it by changing these two values:
```
[[runners]]
   limit = 2
   [[runners-docker]]
      volumes = ["/ opt / docker / gitlab-runner / cache: / cache", "/ dev / shm: / dev / shm"]
```
It seems in our case that it was a question of performance of the machines that performed our jobs, which due to lack of resources killed the docker containers without giving error.
In our gitalb-ci.yml, we update cypress image:
```
test-e2e:
  image: cypress/browsers:node10.16.0-chrome77
```
And this is the result:
![Screenshot-20191016073947-718x162](https://user-images.githubusercontent.com/56622481/66891209-cf3f6400-efe8-11e9-8b0c-75f1556bb3cf.png)
![Screenshot-20191016074103-730x712](https://user-images.githubusercontent.com/56622481/66891214-d4041800-efe8-11e9-9678-6a6b616be9d2.png)

I hope this can help.

**Comment by mlewiarz** [comment]
Hi, we also have the same problem only in Gitlab CI, but I don't think this is related with resources.
Our runners have limit 8CPU and 16GB RAM and there is no sign in logs that cgroups killed this process. 

Anybody was able to fix this problem?

**Comment by jennifer-shehane** [comment]
Since this issue hasn't had activity in a while, we'll close the issue until we can confirm this is still happening. Please comment if there is new information to provide concerning the original issue and we'd be happy to reopen.

**Comment by gelhaimer** [comment]
We've encountered the same problem as mentionned.

![image](https://user-images.githubusercontent.com/3752322/117625154-04d11280-b176-11eb-8c9a-4a3b18da39d5.png)


We use cypress  with cucumber-js and its cypress preprocessor  on an angular projet.

cypress: 4.8.0
cypress-cucumber-preprocessor: 4.1.0
angular: 11.2.1

**Comment by joergmis** [comment]
I also run into the issue that some jobs exit prematurely (see screenshot) with the following setup:

- we tried both cypress 12.9.0 and cypress 12.13.0
- self-hosted gitlab 15.13.0
- gitlab runner 16.x
- docker image `cypress/included:12.13.0`

The jobs look more or less like this:

```yml
before_script:
  - Xvfb :99 &
  - export DISPLAY=:99
script:
  - npx cypress run --browser chrome --spec "xxx"
after_script:
  - pkill Xvfb
```

There are a few things to note:

- this seems to be happening more consistently since we updated the runner configuration to allow running jobs in parallel (with the `concurrent` setting, as mentioned [in the docs](https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-global-section))
- not sure if relevant but after changing the jobs to be run in parallel I also had [to start/stop the `xvfb` server](https://docs.cypress.io/guides/continuous-integration/introduction#Xvfb)
- the job from the screenshot has an `after_script` defined which does not run/show up in the logs (it should kill the `xvfb` server)

<img width="1127" alt="image" src="https://github.com/cypress-io/cypress/assets/26046904/2c2a08e8-86a2-4d8c-be09-832fa6087e72">

**Comment by joergmis** [comment]
> It seems in our case that it was a question of performance of the machines that performed our jobs, which due to lack of resources killed the docker containers without giving error.

I think we were also able to solve this with the `limit` settings @magarlo mentioned. I didn't consider this at the beginning since I'm pretty sure that it's not a resource problem in our case - all runners run on the same machine and overall still have the same amount of concurrent jobs. But anyway, the following config seems to work for now:

```toml
concurrent = 4

# we have 4 runners with the docker executor and this limit setting
[[runner]]
    limit = 1 <-- this is new
```

It would be interesting to know if all people with issues use the docker executor. Maybe it's something like sharing a volume or cache that messes up the tests.
